polygeist/tools/cgeist/Test/Verification/sycl/builtin_types.cpp

```diff
@@ -0,0 +1,61 @@
+// RUN: clang++ -fsycl -fsycl-device-only -O0 -w -emit-mlir -o - %s | FileCheck %s
+
+#include <sycl/sycl.hpp>
+
+using namespace sycl;
+
+// CHECK-LABEL: func.func @_Z13opencl_float2Dv2_f(
+// CHECK:         %arg0: vector<2xf32> {llvm.noundef})
+SYCL_EXTERNAL void opencl_float2(__cl_float2 var) {}
+
+// CHECK-LABEL: func.func @_Z13opencl_float4Dv4_f(
+// CHECK:         %arg0: vector<4xf32> {llvm.noundef})
+SYCL_EXTERNAL void opencl_float4(__cl_float4 var) {}
+
+// CHECK-LABEL: func.func @_Z15scalable_vec2_tN4sycl3_V13vecIfLi2EEE(
+// CHECK:         %arg0: memref<?x!sycl_vec_f32_2_> {llvm.align = 8 : i64, llvm.byval = !sycl_vec_f32_2_, llvm.noundef})
+SYCL_EXTERNAL void scalable_vec2_t(sycl::vec<float, 2> var) {}
+
+// CHECK-LABEL: func.func @_Z15scalable_vec4_tN4sycl3_V13vecIfLi4EEE(
+// CHECK:         %arg0: memref<?x!sycl_vec_f32_4_> {llvm.align = 16 : i64, llvm.byval = !sycl_vec_f32_4_, llvm.noundef})
+SYCL_EXTERNAL void scalable_vec4_t(sycl::vec<float, 4> var) {}
```

Comment:
I meant something like: `__SVFloat32_t`

Reply:
This type is an ARM SVE type. It'd require compiling the code for that arch in the test. If this is not working as is, I suggest adding an `llvm_unreachable` int he case dealing with these types, as we're not targeting this arch and is irrelevant for now.

Reply:
That builtinn type isn't legal on my build machine:

 error: unknown type name '__SVFloat32_t'

Reply:
Added llvm_unreachable for all those ARM types.